aten/src/ATen/core/dispatch/KernelRegistration.h

```diff
@@ -80,58 +156,58 @@ class KernelRegistrar final {
  * @tparam OpSchemaDef The operator schema this is building a KernelRegistration for
  * @tparam FieldsPresentFlags Remembers which fields are already set in the builder
  */
-template<class OpSchemaDef, class StateTypeOrVoid, uint64_t FieldsPresentFlags>
+template<class CacheTypeOrVoid, uint64_t FieldsPresentFlags>
 class KernelRegistrationBuilder final {
 private:
-  using Schema = OpSchema<OpSchemaDef>;
-
   static constexpr uint64_t DISPATCH_KEY_PRESENT = 0x01 << 0;
   static constexpr uint64_t KERNEL_PRESENT = 0x01 << 1;
-  static constexpr uint64_t STATE_PRESENT = 0x01 << 2;
+  static constexpr uint64_t CACHE_PRESENT = 0x01 << 2;
```

Comment:
So, if I understand correctly, this bitfield business is so that you can check if the fields are all OK at compile-time? That would be nice to have a comment to that effect.

Reply:
What do the error messages look like here?

Reply:
Yes, that's correct. Error messages come from the `static_assert`s that are further below.